ricecooker/classes/nodes.py

```diff
@@ -212,6 +216,46 @@ def get_json_tree(self):
 
         return tree
 
+
+    def save_channel_children_to_csv(self, metadata_csv, topic_structure = list()):
+        # Not including channel title in topic structure
+        
+        is_channel = isinstance(self, ChannelNode)
+ 
+        if not is_channel:
+            # Build out tag string
+            current_level = self.title
+            # prevent list from being passed by reference. Create new instance of list
+            structure_list = list(topic_structure)
+            if len(structure_list) < 1:
+                structure_list = [current_level]
+            else:
+                structure_list.append(current_level)
+            structure_list_string = '/'.join(structure_list)
+            tags_string = ','.join(self.tags)
+
+            new_title = self.node_modifications.get('New Title') or ''
+            new_description = self.node_modifications.get('New Description') or ''
+
+            record = [
+                self.source_id,
+                structure_list_string,
+                self.title,
+                self.node_modifications.get('New Title') or '',         # New Title
+                self.description,
+                self.node_modifications.get('New Description') or '',  # New Description
+                tags_string,
+                self.node_modifications.get('New Tags') or '',         # New Tags
+                ''                                                     # Last Modified
+            ]
+            metadata_csv.writerow(record)
+            for child in self.children:
+                child.save_channel_children_to_csv(metadata_csv, structure_list)
+        else:
+            for child in self.children:
+                child.save_channel_children_to_csv(metadata_csv, topic_structure)
```

Comment:
Once we remove the need to have two list variables, this for loop can be moved out of the if / else and always run.

Reply:
I'm still seeing the for loop inside both the if and else blocks. The reason we want to avoid having code that does the same thing in each block is that any time we change one we must remember to also change the other.

Reply:
Sorry about that. Changed to only one for loop